File: C#/Longest Substring Without Repeating Characters/Longest Substring Without Repeating Characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        if(s.Length<=1) return s.Length;
        int len=0,rs=-1;
        var hs=new HashSet<char>();
        for(int i=0;i<s.Length;i++){
            if(i!=0) hs.Remove(s[i-1]);
            if(i>=s.Length-len) break;
            while(rs+1<s.Length && !hs.Contains(s[rs+1])){
                hs.Add(s[rs+1]);
                rs++;
            }
            len=rs-i+1>len?rs-i+1:len;
        }
        return len;
    }
}

File: Python/Longest Substring Without Repeating Characters/Longest Substring Without Repeating Characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l,rs,n=0,-1,len(s)
        hs=set()
        for i in range(n):
            if i!=0:
                hs.remove(s[i-1])
            while rs+1<n and s[rs+1] not in hs:
                hs.add(s[rs+1])
                rs+=1
            l=max(l,rs-i+1)
        return l